### PR TITLE
Add solver for first order acoustic wave equation

### PR DESCRIPTION
**Pull request by acitrain**
This PR add a solver for the first order wave equation.

The equation are solved using Spectral Element Method in space and 2nd order Leap-Frog in time
The main difference with the second order equation are the following:
- Two unknowns (pressure and velocity) insterad of one
- The pressure is considered continuous and the velocity discontinuous
- The stiffness matrix is the product of a gradient and a basis function instead of two gradient. The computation of the stiffness values is optimize as done for the second order equation in PR #2123.

Rebaseline done in https://github.com/GEOSX/integratedTests/pull/298

### CODE REVIEW

**Review by sframba — APPROVED** [review]

**Comment by sframba** [comment]
We could a unit test to `testWavePropagation.cpp`, it can simply be a copy-paste of the one for the second-order formulation

File: src/coreComponents/physicsSolvers/wavePropagation/AcousticFirstOrderWaveEquationSEMKernel.hpp [hunk 1]
```diff
@@ -0,0 +1,583 @@
...
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file AcousticFirstOrderWaveEquationSEMKernel.hpp
+ */
+
+#ifndef GEOSX_PHYSICSSOLVERS_WAVEPROPAGATION_AcousticFirstOrderWaveEquationSEMKERNEL_HPP_
+#define GEOSX_PHYSICSSOLVERS_WAVEPROPAGATION_AcousticFirstOrderWaveEquationSEMKERNEL_HPP_
```
Comment (francoishamon) [inline src/coreComponents/physicsSolvers/wavePropagation/AcousticFirstOrderWaveEquationSEMKernel.hpp:20]:
```suggestion
#ifndef GEOSX_PHYSICSSOLVERS_WAVEPROPAGATION_ACOUSTICFIRSTTORDERWAVEEQUATIONSEMKERNEL_HPP_
#define GEOSX_PHYSICSSOLVERS_WAVEPROPAGATION_ACOUSTICFIRSTTORDERWAVEEQUATIONSEMKERNEL_HPP_
```

**Review by francoishamon — APPROVED** [review]
Looks great! I just made some comments to see if it is possible to avoid the code duplication introduced by this PR.

**Comment by francoishamon** [comment]
I checked that the integrated tests are passing with this branch on Quartz and Lassen. After coordinating with Aurelien and Stefano, I will be merging this PR when the unit tests have passed.

File: src/coreComponents/physicsSolvers/wavePropagation/WaveSolverBase.cpp [hunk 2]
```diff
@@ -140,6 +172,22 @@ void WaveSolverBase::reinit()
 void WaveSolverBase::initializePreSubGroups()
 {
   SolverBase::initializePreSubGroups();
+
+  localIndex const numNodesPerElem = WaveSolverBase::getNumNodesPerElem();
+
+  std::cout << numNodesPerElem << std::endl;
```
Comment (francoishamon) [inline src/coreComponents/physicsSolvers/wavePropagation/WaveSolverBase.cpp:178]: @acitrain maybe remove in a later PR

File: src/coreComponents/physicsSolvers/wavePropagation/WaveSolverBase.cpp [hunk 3]
```diff
@@ -140,6 +172,22 @@ void WaveSolverBase::reinit()
...
+
+  localIndex const numSourcesGlobal = m_sourceCoordinates.size( 0 );
+  std::cout << numSourcesGlobal << std::endl;
```
Comment (francoishamon) [inline src/coreComponents/physicsSolvers/wavePropagation/WaveSolverBase.cpp:181]: Maybe remove in a later PR

File: src/coreComponents/physicsSolvers/wavePropagation/AcousticFirstOrderWaveEquationSEMKernel.hpp [hunk 4]
```diff
@@ -0,0 +1,584 @@
...
+                                     sourceCoordinates[isrc][1],
+                                     sourceCoordinates[isrc][2] };
+
+          bool const sourceFound =
+            WaveSolverUtils::locateSourceElement( numFacesPerElem,
+                                                  center,
+                                                  faceNormal,
+                                                  faceCenter,
+                                                  elemsToFaces[k],
+                                                  coords );
```
Comment (francoishamon) [inline src/coreComponents/physicsSolvers/wavePropagation/AcousticFirstOrderWaveEquationSEMKernel.hpp:117]: In a later PR, we should remove this function and use the one available in `ComputationalGeometry.hpp` to benefit from the improvements made by @untereiner and @dkachuma

**Review by francoishamon — COMMENTED** [review]
Looks great

**Pull request merged**